src/org/labkey/test/components/ui/grids/QueryGrid.java

```diff
@@ -391,4 +447,104 @@ protected Locator locator()
             return _locator;
         }
     }
+
+    /**
+     * Dialog that will allow a user to save a grid view.
+     */
+    public static class SaveViewDialog extends ModalDialog
+    {
+        private final Input viewNameInput = Input.Input(Locator.name("gridViewName"), getDriver())
+                .findWhenNeeded(this);
+
+        Checkbox checkbox = new Checkbox(Locator.input("setDefaultView").findWhenNeeded(this));
+
+        public SaveViewDialog(WebDriver driver)
+        {
+            super(new ModalDialogFinder(driver).withTitle("Save Grid View"));
+        }
+
+        /**
+         * Set the view name.
+         *
+         * @param viewName View name.
+         * @return This dialog.
+         */
+        public SaveViewDialog setViewName(String viewName)
+        {
+            viewNameInput.set(viewName);
+            return this;
+        }
+
+        /**
+         * Get the value of the View Name field.
+         *
+         * @return Value of View Name field.
+         */
+        public String getViewName()
+        {
+            return viewNameInput.get();
+        }
+
+        /**
+         * Check if the View Name field is enabled. It should be disabled if the 'Make default for all' checkbox is checked.
+         *
+         * @return True if enabled, false otherwise.
+         */
+        public boolean isViewNameEnabled()
+        {
+            return viewNameInput.getComponentElement().isEnabled();
+        }
+
+        /**
+         * Check or uncheck the 'Make default view' checkbox.
+         *
+         * @param checked True to check, false to uncheck.
+         * @return This dialog.
+         */
+        public SaveViewDialog setMakeDefaultForAll(boolean checked)
+        {
+            if(checked)
+            {
+                checkbox.check();
+            }
+            else
+            {
+                checkbox.uncheck();
+            }
+
+            return this;
+        }
+
+        /**
+         * Get the checked status of the 'Make default for all' checkbox.
+         *
+         * @return True if it is checked, false otherwise.
+         */
+        public boolean isMakeDefaultForAllChecked()
+        {
+            return checkbox.isChecked();
+        }
+
+        /**
+         * Save the view.
+         */
+        public void saveView()
+        {
+            dismiss("Save", 1);
+        }
```

Comment:
If you made this class non-static, you could wait for some state change in the `QueryGrid` when clicking save.
```suggestion
        public void saveView()
        {
            dismiss("Save", 1);
            // Wait for save button on the grid to disappear or something like that
            // e.g.
            waitFor(() -> !Locator.buttonContainingText("Save").findWhenNeeded(QueryGrid.this.getComponentElement()).isDisplayed(), 5_000);
        }
```

Reply:
I could but it feels like that is something that would be more appropriate to be done in ModalDialog.dismiss(String bittonText).

Reply:
Following up on this.
This code has changed. In the apps modifying a grid view and then saving it is different from LKS. In the app modifying the grid and saving the view are two different dialogs. Modifying the gird will result in an update to the grid, and that dialog does take a reference to the grid and waits for it to update. Saving a grid view does not result in a grid update and will not need to wait for it.